src/test/java/com/restapi/carMarket/service/CarServiceUnitTest.java

```diff
@@ -11,168 +8,199 @@
         Mockito.verify(carDAO,Mockito.times(0)).save(car);
     }
 
     @Test(expected = CarNotValidException.class)
-    public void insertionOfCarWithPriceLessThan1ShouldThrowCarNotValidException() throws Exception{
+    public void insertionOfCarWithPriceLessThan1ShouldThrowCarNotValidException() {
         Car car = new Car("BMW","5 Series", 2010, 0);
-        carService.insert(car);
+        carService.addCarToMarket(car);
         Mockito.verify(carDAO,Mockito.times(0)).save(car);
     }
 
     @Test(expected = CarNotValidException.class)
-    public void insertionOfCarWithPriceToHighShouldThrowCarNotValidException() throws Exception{
+    public void insertionOfCarWithPriceToHighShouldThrowCarNotValidException() {
         Car car = new Car("BMW","5 Series", 2010, 2100000000);
-        carService.insert(car);
+        carService.addCarToMarket(car);
+        Mockito.verify(carDAO,Mockito.times(0)).save(car);
+    }
+
+    @Test(expected = CarNotValidException.class)
+    public void insertionOfCarWithBlankBrandShouldThrowCarNotValidException() {
+        Car car = new Car("","5 Series", 2010, 2100000);
+        carService.addCarToMarket(car);
+        Mockito.verify(carDAO,Mockito.times(0)).save(car);
+    }
+
+    @Test(expected = CarNotValidException.class)
+    public void insertionOfCarWithNullBrandShouldThrowCarNotValidException() {
+        Car car = new Car(null,"5 Series", 2010, 2100000);
+        carService.addCarToMarket(car);
+        Mockito.verify(carDAO,Mockito.times(0)).save(car);
+    }
+
+    @Test(expected = CarNotValidException.class)
+    public void insertionOfCarWithBlankModelShouldThrowCarNotValidException() {
+        Car car = new Car("BMW","", 2010, 2100000);
+        carService.addCarToMarket(car);
+        Mockito.verify(carDAO,Mockito.times(0)).save(car);
+    }
+
+    @Test(expected = CarNotValidException.class)
+    public void insertionOfCarWithNullModelShouldReturnCarNotValidException() {
+        Car car = new Car("BMW",null, 2010, 2100000);
+        carService.addCarToMarket(car);
         Mockito.verify(carDAO,Mockito.times(0)).save(car);
     }
 
     @Test
-    public void findAllCarsShouldCallFindAllMethodOfDAO() throws Exception{
-        carService.findAll();
+    public void findAllCarsShouldCallFindAllMethodOfDAO() {
+        carService.getAllCarsOnMarket();
         Mockito.verify(carDAO,Mockito.times(1)).findAll();
     }
 
     @Test
-    public void findExistentCarByIdShouldReturnThisCar() throws Exception{
+    public void findExistentCarByIdShouldReturnThisCar() {
         Car car = new Car("BMW","5 Series", 2200, 40000);
         Optional<Car> returnedCar = Optional.of(car);
         Long id = 10L;
 
         given(carDAO.findById(id)).willReturn(returnedCar);
-        Car newCar = carService.findById(id);
+        Car newCar = carService.getCarById(id);
 
         Assert.assertTrue(newCar.equals(car));
         Mockito.verify(carDAO,Mockito.times(1)).findById(id);
```

Comment:
No need to verify this call because it's stubbed on line 109. Verify is used to verify side effects, i.e. method returning `void` or those calls whose return value is ignored. In your case result of `carDao.findById(id)` is not ignored but instead is returned back to this test case. Therefore, if `newCar` is the `car` then we know the dao was called.

Reply:
I see, missed that logic. Done.